### DISCUSSION TITLE
Jetbrains IDE does not persist/recognize a succesful login

### DISCUSSION THREAD

**Discussion by fatso83** [community/community#35422] [Copilot Conversations]
I am unable to use Copilot, as every time I succesfully complete the sing and dance of adding my "device" (whatever that is supposed to mean), the status of Github Copilot does not change. It still thinks I am not logged in.

I start out here:
![image](https://user-images.githubusercontent.com/618076/194535737-27d875de-83fc-43c4-8586-405b4788e4f0.png)
Then go through adding the code in the coming prompt, waiting for the callback to IDEA to complete (i.e. the dialogue window about waiting for authentication to close), and then I check the status of Copilot and it never changes from this:

![image](https://user-images.githubusercontent.com/618076/194535662-e9ce2a58-4b12-4e91-9392-442bc4d95440.png)

I have now gone through this process 4 times with no apparent change. The IDE has not thrown any exceptions that I am aware of (at least not notifying me about it), so I am not sure what to do from here. 

P.S. Tried finding similar issues, but [I am not using a proxy](https://github.com/orgs/community/discussions/16828)

**Comment by iamcjwilliams** [comment]
@fatso83 Our copilot team is reviewing this issue

**Reply by fatso83** [reply]
In case you need logs of some kind, more details, etc. to recognize the issue, just let me know and if I should use some non-public channel to avoid disclosing sensitive data. Not sure what else I can do atm.

**Reply by iamcjwilliams** [reply]
@fatso83 please provide any logs and/or details, this will help with identifying the issue quicker

**Reply by fatso83** [reply]
I would if I could, but as mentioned there were no exceptions popping up. So if I were to gather some debug logs from somewhere I would need to know how.

> Not sure what else I can do atm

**Reply by bmuskalla** [reply]
Logs can be found in the usual place for Intellij: https://intellij-support.jetbrains.com/hc/en-us/articles/207241085-Locating-IDE-log-files

**Reply by fatso83** [reply]
Thanks for that, but there was absolutely nothing interesting I could find in that log after logging in. After being authorized to GitHub in the web browser the getting started guide said I was to expect a confirmation dialog about being successfully logged in. That never showed up. Instead the "waiting for authorization" dialog just closed. Not sure if this is a hint.

Here's the log:
[idea.log](https://github.com/community/community/files/9765839/idea.log)

**Reply by bmuskalla** [reply]
It has several errors regarding `Invalid copilot token: missing token: 403`. Can you check what https://github.com/settings/copilot says?

**Reply by fatso83** [reply]
That was it. There was no subscription setup or initial configuration. 
<img width="1552" alt="Skjermbilde 2022-10-12 kl  21 37 58" src="https://user-images.githubusercontent.com/618076/195434994-70f274b8-44ab-459c-b6cb-8c42facffc71.png">
I then accepted. Still no go.
<img width="814" alt="Skjermbilde 2022-10-12 kl  21 47 01" src="https://user-images.githubusercontent.com/618076/195434991-fe61e0a6-91f9-41d9-bdcd-297f3deba609.png">
Then read the [quick start docs][quick-start] and figured I need to choose yes or no for public code usage
<img width="321" alt="Skjermbilde 2022-10-12 kl  21 49 00" src="https://user-images.githubusercontent.com/618076/195434988-d6eefd55-9c81-43ad-930e-a24f571c64d0.png">

This is what the [official quick start][quick-start] asks one to do, so quite expected _if one comes through that_. That assumption is invalid for people coming to Copilot via IntelliJ's plugin install and so should be had a look at.
1. The [Getting started with GitHub Copilot in a JetBrains IDE](https://docs.github.com/en/copilot/getting-started-with-github-copilot/getting-started-with-github-copilot-in-a-jetbrains-ide#prerequisites) does not list or mention this at all, not even in the prerequisites.
2. The IDE plugin should know that something is off, yet does not alert the user

So that is two things that could avoid users contacting you needlessly 😄 Thanks for the help.

[quick-start]: https://docs.github.com/en/copilot/quickstart#signing-up-for-github-copilot
